orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/PipelineRefTrigger.kt

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.sql.pipeline.persistence
+
+import com.netflix.spinnaker.kork.artifacts.model.Artifact
+import com.netflix.spinnaker.kork.artifacts.model.ExpectedArtifact
+import com.netflix.spinnaker.orca.api.pipeline.models.PipelineExecution
+import com.netflix.spinnaker.orca.api.pipeline.models.Trigger
+import com.netflix.spinnaker.orca.pipeline.model.PipelineTrigger
+
+/**
+ * A trigger type used by SQL implementations to store a [PipelineTrigger] without embedding the entire parent
+ * [PipelineExecution]. Instead of embedding, only the execution ID is stored, and then eagerly loaded at
+ * deserialization time.
+ */
+data class PipelineRefTrigger(
```

Comment:
One thing I really dont like is if our pipeline structure changed, we would have to remember to update this as well, right? We should re-think about this if that's the case, which I think it is. This is one area I can see a bug happening

Reply:
I see what you mean but the PipelineRefTrigger extends of the main Trigger class so any change in the base trigger class shouldn't break PipelineRefTrigger. An option that comes to my mind is I can try to extend from PipelineTrigger class but I am not seeing a real benefit on doing that.

I would say the best option will be update the structure of the PipelineTrigger to have the `parentExecutionId` but that will be a big change on the behavior and the feature flag would be useless.

I am open to option you have in mind as well :)

Reply:
Hmm, let me think about this some and Ill get back to you on this. I want to look into the various options. I definitely dont want to block this PR if a major change is needed. However, I want to first exhaust a few options before moving forward. I may need a couple days, if that's okay, or if some other reviewer could do it, then Id be okay with that as well.

Reply:
Okay, after looking I think the best approach may be to wrap the `PipelineTrigger` class so we don't need to maintain two separate pipeline models.

My thoughts were, mind you I did not spend too much thinking about how valid this would be
```java
data class PipelineRefTrigger(
    private val pipelineTrigger: PipelineTrigger // The change I care about most is here
    // then whatever relevant references here
```

This should simplify this class tremendously and allow us to only need to manage one model. Further with the object mapper change, this becomes pretty easy to handle as well :)

Reply:
For context, this was closed cause it turned out that if we add the `PipelineTrigger` as a field, it causes the deserializer to not work properly. Rather than trying to find out all the key details, Im fine with having it as is, and fixing this later with some jackson magic to keep things backwards compatible.